pkg/skaffold/build/cache/hash_test.go

```diff
@@ -146,17 +148,102 @@ func TestGetHashForArtifact(t *testing.T) {
 	}
 	for _, test := range tests {
 		testutil.Run(t, test.description, func(t *testutil.T) {
-			t.Override(&hashFunction, mockCacheHasher)
-			t.Override(&artifactConfigFunction, fakeArtifactConfig)
+			t.Override(&fileHasherFunc, mockCacheHasher)
+			t.Override(&artifactConfigFunc, fakeArtifactConfig)
 			if test.artifact.DockerArtifact != nil {
 				tmpDir := t.NewTempDir()
 				tmpDir.Write("./Dockerfile", "ARG SKAFFOLD_GO_GCFLAGS\nFROM foo")
 				test.artifact.Workspace = tmpDir.Path(".")
-				test.artifact.DockerArtifact.DockerfilePath = "Dockerfile"
+				test.artifact.DockerArtifact.DockerfilePath = Dockerfile
 			}
 
 			depLister := stubDependencyLister(test.dependencies)
-			actual, err := getHashForArtifact(context.Background(), depLister, test.artifact, test.mode)
+			actual, err := newArtifactHasher(nil, depLister, test.mode).hash(context.Background(), test.artifact)
+
+			t.CheckNoError(err)
+			t.CheckDeepEqual(test.expected, actual)
+		})
+	}
+}
+
+func TestGetHashForArtifactWithDependencies(t *testing.T) {
+	tests := []struct {
+		description  string
+		dependencies map[string][]string
+		artifacts    []*latest.Artifact
+		mode         config.RunMode
+		expected     string
+	}{
+		{
+			description:  "hash for artifact with two dependencies",
+			dependencies: map[string][]string{"img1": {"a"}, "img2": {"b"}, "img3": {"c"}, "img4": {"d"}},
```

Comment:
This is hard to understand: there is no "a", "b", "c", or "d" visible.  There should be a comment above to describe what these mean.